### DISCUSSION TITLE
What is the status of the rewrite?

### DISCUSSION THREAD

**Discussion by ghost** [TeamNewPipe/NewPipe#11676] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my question isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the question will be dismissed otherwise.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### What is/are your question(s)?

What is the status of the rewrite?
Can I see the state of it? Can I follow the progress?

**Comment by asandikci** [comment]
see refactor branch

**Reply by ghost** [reply]
No actions for 2 weeks now

**Reply by asandikci** [reply]
Oh didn't see that. dev branch is active but probably changes are not related with refactor. I also wonder about current status of refactor

**Reply by ShareASmile** [reply]
As rightly pointed out here, The rewrite is happening on the [refactor branch](https://github.com/TeamNewPipe/NewPipe/tree/refactor).
And
We have a [project](https://github.com/orgs/TeamNewPipe/projects/18) to track/manage the Rewrite.

**Reply by ghost** [reply]
Is the rewriting still active? No actions for 2 weeks and nearly no actions at all.

**Reply by ShareASmile** [reply]
Yes, it is active. As it is a continuing process which may seem to be slow at times.

**Reply by ghost** [reply]
Hi, is rewrite still going on?

The [project](https://github.com/orgs/TeamNewPipe/projects/18/views/1) has not been updated for a few months now.

**Comment by ShareASmile** [comment, marked as answer]
> Hi, is rewrite still going on?

> The [project](https://github.com/orgs/TeamNewPipe/projects/18/views/1) has not been updated for a few months now.

Yes, please take a look at [PRs with rewrite label](https://github.com/TeamNewPipe/NewPipe/pulls?q=is%3Aopen+is%3Apr+label%3Arewrite) particularly at this imporant one, it is in review process [initial work to integrate NewPlayer into NewPipe](https://github.com/TeamNewPipe/NewPipe/pull/11965)

BTW, you might want to take a look at #12060 & read our [latest blogpost](https://newpipe.net/blog/pinned/announcement/newpipe-0.27.6-rewrite-team-states/) about latest release, state of the rewite, team status & difficulties team is facing due to shortage of manpower & dearth of free time team members can spend on it due to their personal obligations such as studies, university assignments etc. 

We hired Schabi for complete rewrite of new player based on Media3 last year for 6 months and he delivered great implementation of NewPlayer in a [separate repository](https://github.com/TeamNewPipe/NewPlayer) and then hired Profpatsch who picked up Schabi’s work and is integrating NewPlayer into NewPipe. He is doing great job but as said due to less time of team members review process is kind of slow but it's picking up. Important thing as blogpost dwells upon is "We do need more developers for writing code, to review PRs & maintainers from the community to do it swiftly & smoothly."

Further on you can comment on related topics raised in blogpost in the linked discussion #12059.

**Reply by ghost** [reply]
I like to hear that, thank you!
